functionary/prompt.py

```diff
@@ -37,12 +37,27 @@ def from_message(cls, message: Dict) -> EndToken:
         elif role == "function":
             return EndToken.function
         else:  # role = assistant
-            if "function_call" in message:
+            if "function_call" in message and message["function_call"] is not None:
                 return EndToken.function_call
             else:
                 return EndToken.assistant
 
 
+class BeginToken(str, Enum):
```

Comment:
It should be named as "StartToken" as the exact opposite of end is "start"

Reply:
sure